jujugui/static/gui/src/test/utils.js

```diff
@@ -1,204 +1,189 @@
+    this.onclose = function() {};
+    this.onmessage = function() {};
+    this.onopen = function() {};
 
-    },
+  },
 
-    getter: function(attributes, default_) {
-      return function(name) {
-        if (attributes.hasOwnProperty(name)) {
-          return attributes[name];
-        } else {
-          return default_;
+  getter: function(attributes, default_) {
+    return function(name) {
+      if (attributes.hasOwnProperty(name)) {
+        return attributes[name];
+      } else {
+        return default_;
+      }
+    };
+  },
+
+  setter: function(attributes) {
+    return function(name, value) {
+      attributes[name] = value;
+    };
+  },
+
+  /**
+   * Util to load a fixture (typically as 'data/filename.json').
+   *
+   * @method loadFixture
+   * @param {String} url to synchronously load, appended /test/ base url.
+   * @param {Boolean} parseJSON when true return will be processed
+   *                  as a JSON blob before returning.
+   * @return {Object} fixture data resulting from call.
+   */
+  loadFixture: function(url, parseJson) {
+    var tries = 3;
+    var response;
+    url = GlobalConfig.test_url + url;
+    while (true) {
+      try {
+        const request = new XMLHttpRequest();
+        request.open('GET', url, false);
+        request.send(null);
+        response = request.responseText;
+        if (parseJson) {
+          response = JSON.parse(response);
         }
-      };
-    },
-
-    setter: function(attributes) {
-      return function(name, value) {
-        attributes[name] = value;
-      };
-    },
-
-    /**
-     * Util to load a fixture (typically as 'data/filename.json').
-     *
-     * @method loadFixture
-     * @param {String} url to synchronously load, appended /test/ base url.
-     * @param {Boolean} parseJSON when true return will be processed
-     *                  as a JSON blob before returning.
-     * @return {Object} fixture data resulting from call.
-     */
-    loadFixture: function(url, parseJson) {
-      var tries = 3;
-      var response;
-      url = GlobalConfig.test_url + url;
-      while (true) {
-        try {
-          const request = new XMLHttpRequest();
-          request.open('GET', url, false);
-          request.send(null);
-          response = request.responseText;
-          if (parseJson) {
-            response = JSON.parse(response);
-          }
-          break;
-        } catch (e) {
-          tries -= 1;
-          if (tries <= 0) {
-            throw e;
-          }
+        break;
+      } catch (e) {
+        tries -= 1;
+        if (tries <= 0) {
+          throw e;
         }
       }
-      return response;
     }
-  };
+    return response;
+  }
+};
 
+YUI(GlobalConfig).add('juju-tests-utils', function(Y) {
+  var jujuTests = Y.namespace('juju-tests');
+  jujuTests.utils = testUtils;
 }, '0.1.0', {
-  requires: [
-    'io',
-    'node',
-    'promise',
-    'datasource-local'
-  ]
+  requires: []
 });
+
+module.exports = testUtils;
```

Comment:
Expose the module as well as allowing it to be included with YUI until all the tests are updated.